### DISCUSSION TITLE
How can I Use Self-Signed Certificates in StackStorm Sensor

### DISCUSSION THREAD

**Discussion by tikh-iv** [StackStorm/st2#6228] [Operations]
Hello,

I am encountering an issue when trying to use a self-signed certificate in a StackStorm sensor. Below are the details of my setup and the error I am facing

## Environment:
- StackStorm Version: 3.8.0
- Python Version: 3.8.17

## Steps:
1. I’ve added my self-signed certificate to the following path:
`/opt/stackstorm/st2/lib/python3.8/site-packages/certifi/cacert.pem`
2. I am trying to run the sensor for testing using the following command:
`sudo /opt/stackstorm/st2/bin/st2sensorcontainer --config-file=/etc/st2/st2.conf --sensor-ref=test_sensor.test_sensor`

## Issue:
Despite adding my self-signed certificate, I am encountering the following error
`2024-07-30 18:43:12,881 ERROR [-] Sensor "test_sensor" run method raised an exception: HTTPSConnectionPool(host='test_sensor.test.local', port=443): Max retries exceeded with url: /v1/api/test_sensor (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate in certificate chain (_ssl.c:1131)'))).`

## Question:
How can I properly configure StackStorm or the sensor to use my self-signed certificates? Are there specific settings or additional steps required to ensure that the self-signed certificate is recognized?

**Comment by chain312** [comment]
Do you have to use a certificate? When I encounter https, I will directly verify=False
`response = requests.request("POST", url, headers=headers, data=payload, verify=False)`
